### PR TITLE
mruby-cli app can be released with task release

### PR DESCRIPTION
**Pull request by toch**
@hone @zzak here the results of our discussion. Thanks a lot, I find the result much cleaner than all the hacks and workarounds I can suggest :stuck_out_tongue: What do you think of this version?

Don't you think we should document which attributes should be set outside of the blocks?

### CODE REVIEW

**Comment by toch** [comment]
ping?

File: Rakefile [hunk 1]
```diff
@@ -19,6 +19,10 @@ Rake::Task[:mruby].invoke unless Dir.exist?(mruby_root)
 Dir.chdir(mruby_root)
 load "#{mruby_root}/Rakefile"
 
+load "#{mruby_root}/../mrbgem.rake"
```
Comment (zzak) [inline Rakefile]: I _think_ it's safe to assume that our `mrbgem.rake` will always be next to `Rakefile`, while `mruby_root` may not always be one directory down from our source dir.

File: mrblib/mruby-cli/setup.rb [hunk 2]
```diff
@@ -386,11 +393,43 @@ def clean_env(envs)
   sh "rake deep_clean"
 end
 
+desc "generate a release tarball"
+task :release => :compile do
```
Comment (zzak) [inline mrblib/mruby-cli/setup.rb:400]: I'd like to merge your changes and test it more before we make the `:release` task public.

Once we're happy with it, we should be able to just flip the switch and add the task to setup.

File: Rakefile (hunk 1 above)
Comment (hone) [inline Rakefile]: Yeah, that sounds right. @toch and I will do `File` stuff to load based off the current `Rakefile`

File: mrblib/mruby-cli/setup.rb (hunk 2 above)
Comment (hone) [inline mrblib/mruby-cli/setup.rb:400]: I think it'll be fine. This PR is a refactoring of the version stuff, but with a fallback if we can't detect the version number I think we'll be good.

**Pull request merged**